lib/convection/control/stack.rb

```diff
@@ -432,7 +434,15 @@ def watch(poll = 2, &block)
       #   the call to ec2 client's describe availability zones
       def availability_zones(&block)
         @availability_zones ||=
-          @ec2_client.describe_availability_zones.availability_zones.map(&:zone_name).sort
+          @ec2_client.describe_availability_zones.availability_zones.map(&:zone_name)
+
+        unless @exclude_availability_zones.empty?
+          @exclude_availability_zones.each { |az| @availability_zones.delete(az) }
+        end
+        if @availability_zones.empty? && block
```

Comment:
It allows me to test removing all of the azs without an exception.  I can remove it and refactor the test :man_shrugging: